src/overlays/actors/ovl_En_Jj/z_en_jj.c

```diff
@@ -53,29 +67,260 @@ static ColliderCylinderInit D_80A88CB4 = {
+                EnJj_SetupAction(this, EnJj_WaitToOpenMouth);
+            } else {
+                EnJj_SetupAction(this, EnJj_WaitForFish);
+            }
+
+            this->bodyCollisionActor = (DynaPolyActor*)Actor_SpawnAsChild(
+                &globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_EN_JJ, this->dyna.actor.world.pos.x - 10.0f,
+                this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z, 0, this->dyna.actor.world.rot.y, 0, JABUJABU_COLLISION);
+            DynaPolyActor_Init(&this->dyna, 0);
+            CollisionHeader_GetVirtual(&gJabuJabuHeadCol, &colHeader);
+            this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+            Collider_InitCylinder(globalCtx, &this->collider);
+            Collider_SetCylinder(globalCtx, &this->collider, &this->dyna.actor, &sCylinderInit);
+            this->dyna.actor.colChkInfo.mass = MASS_IMMOVABLE;
+            break;
+
+        case JABUJABU_COLLISION:
+            DynaPolyActor_Init(&this->dyna, 0);
+            CollisionHeader_GetVirtual(&gJabuJabuBodyCol, &colHeader);
+            this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+            func_8003ECA8(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+            this->dyna.actor.update = EnJj_UpdateStaticCollision;
+            this->dyna.actor.draw = NULL;
+            Actor_SetScale(&this->dyna.actor, 0.087f);
+            break;
+
+        case JABUJABU_UNUSED_COLLISION:
+            DynaPolyActor_Init(&this->dyna, 0);
+            CollisionHeader_GetVirtual(&gJabuJabuUnusedCol, &colHeader);
+            this->dyna.bgId = DynaPoly_SetBgActor(globalCtx, &globalCtx->colCtx.dyna, &this->dyna.actor, colHeader);
+            this->dyna.actor.update = EnJj_UpdateStaticCollision;
+            this->dyna.actor.draw = NULL;
+            Actor_SetScale(&this->dyna.actor, 0.087f);
+            break;
+    }
+}
+
+void EnJj_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnJj* this = THIS;
+
+    switch (this->dyna.actor.params) {
+        case JABUJABU_MAIN:
+            DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+            Collider_DestroyCylinder(globalCtx, &this->collider);
+            break;
+            
+        case JABUJABU_COLLISION:
+        case JABUJABU_UNUSED_COLLISION:
+            DynaPoly_DeleteBgActor(globalCtx, &globalCtx->colCtx.dyna, this->dyna.bgId);
+            break;
+    }
+}
+
+/**
+ * Blink routine. Blinks at the end of each randomised blinkTimer cycle. If extraBlinkCounter is not zero, blink that many more times before resuming random blinkTimer cycles. 
+ * extraBlinkTotal can be set to a positive number to blink that many extra times at the end of every blinkTimer cycle, but the actor always sets it to zero, so only one multiblink happens when extraBlinkCounter is nonzero.
+ */
```

Comment:
could be mistaken but i think this line is longer than our 120 limit. run clang if you havent